unlock_indicator.c

```diff
@@ -291,10 +291,29 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
     }
 
     if (xr_screens > 0) {
+        /* This keeps track of the screens drawn to */
+        bool drawn_to[xr_screens];
+        memset(drawn_to, false, sizeof(bool) * xr_screens);
+
         /* Composite the unlock indicator in the middle of each screen. */
         for (int screen = 0; screen < xr_screens; screen++) {
+            bool draw_indicator = true;
+
             int x = (xr_resolutions[screen].x + ((xr_resolutions[screen].width / 2) - (button_diameter_physical / 2)));
             int y = (xr_resolutions[screen].y + ((xr_resolutions[screen].height / 2) - (button_diameter_physical / 2)));
+
+            for (int other = 0; other < screen; other++) {
```

Comment:
Can you add a comment above the loop to make it clear what you are intending to do? Maybe something like:

```c
/* Don’t display the unlock indicator if the middle of the indicator would end
   up on a different display. Prevents multiple indicators in mirrored screen
   configurations (e.g. with 1280x800+0+0 and 1024x768+0+0). */
```